xdsl/dialects/builtin.py

```diff
@@ -43,7 +43,7 @@ def verify(self, attr: Attribute) -> None:
 
 @irdl_attr_definition
 class ArrayAttr(GenericData[List[_ArrayAttrT]]):
-    name: str = "array"
+    name: str = "array_attr"
```

Comment:
?

Reply:
The `ArrayAttr` is a special attribute, it is never actually printed with the name, but always in brackets. I had to rename it because the name clashed with `DenseArrayBase`.